language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Function.hs

```diff
@@ -247,3 +250,42 @@ getBuiltinFixN n = do
         TyAbs () s (Type ()) $
         LamAbs () h hTy $
         mkIterApp () (Var () h) branches
+
+data Function funName ann = Function
+    { _functionAnn :: ann
+    , _functionDom :: Type TyName ann
+    , _functionCod :: Type TyName ann
+    , _functionDef :: Def funName (Term TyName Name ann)
+    }
+
+functionType :: Function name ann -> Type TyName ann
+functionType (Function ann dom cod _) = TyFun ann dom cod
+
+functionVarDecl :: Function (Name ann) ann -> VarDecl TyName Name ann
+functionVarDecl fun@(Function ann _ _ (Def name _)) = VarDecl ann name $ functionType fun
+
+getBuiltinMutualFixOf :: ann -> [Function (Name ann) ann] -> Quote (Tuple ann)
```

Comment:
This is now also quite "meta". I only mention this because we do distinguish them in the module structure, but I'm not sure how useful that is.

Reply:
Sure, just good to bear in mind.